app/controllers/api/v1/holidays_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Api::V1::HolidaysController < Api::V1::BaseController
+  skip_before_action :doorkeeper_authorize!, only: :index
+  expose :holiday,  -> { Holiday.find(params[:id]) }
+  expose :holidays, -> { Holiday.all }
+
+  def index
+    render_api(holidays, :ok, each_serializer: HolidaySerializer)
+  end
+
+  def create
+    return render_api(holidays, :created) if holidays.create(holiday_params)
+    render json: { errors: holidays.errors.messages }, status: :unprocessable_entity
+  end
+
+  def update
+   return render_api(:accepted) if holiday.update(holiday_params)
+   rails_error
+  end
+
+  def destroy
+    return render_api(:ok) if holiday.destroy
+    rails_error
+  end
+
+  def show
+   return render_api(holiday, :ok, each_serializer: HolidaySerializer) if holiday
+   render json: { errors: 'Record not found' }, status: 404
+  end
+
+  private
+
+  def rails_error
+    render json: { errors: holiday.errors.messages }, status: :unprocessable_entity
```

Comment:
I don't think it makes sense to extract this single row to a separate method.

Reply:
def rails error removed